Keithley2612B_voltage_sweep.py

```diff
@@ -164,36 +235,41 @@ def sweep_operation(smu_id, steps_no, measure_delay, nplc, start, end):
 
             target_volt = get_target_volt(start_volt)
             
-            steps_num = int (input ('Enter the number of steps'))
+            steps_num = int (input ('Enter the number of steps: '))
 
             integration_time = get_integration_time()
             
-            measure_delay = float (input('Set settling delay (in seconds) before each measurement: (NOTE: Setting the value to -1 automatically starts taking measurement as soon as current is stable) '))
+            measure_delay = 0
 
             #pulsed = get_sweep_type()
 
+            scan_rate = float(input ('Enter scan rate in mV/s: '))
             
             # VI measurement
             print ('Voltage sweep being conducted...')
-            test_output = sweep_operation (smu, steps_num, measure_delay, integration_time, start_volt, target_volt) 
+            test_output = sweep_operation (smu, steps_num, measure_delay, integration_time, start_volt, target_volt, scan_rate) 
             end_time = datetime.now()
             print (f"Sweep successfully completed on {end_time}.")
 
             # scan rate 
             # (Unsure of the data type of measured voltage and timestamps. This will only work if both are float or int)   
     
-            del_t = []
+            del_t = np.empty((1,steps_num))
+            print("The test_output value is: " + str(test_output[0]))
+            print("The length is: " + str(len(test_output[2])))
             for i in range(len(test_output[2])):
-                del_t [i] = test_output [2][i] - test_output [2][0]
+                del_t[0][i] = test_output[2][i] - test_output[2][0]
+
+            print("del t = " + str(del_t))
 
-            scan_rate = np.polyfit(del_t, test_output[0], 1)[0]
+            scan_rate = np.polyfit(del_t[0], test_output[0], 1)
```

Comment:
I added [0] at the end of the polyfit function because linear polyfit returns an array of the slop and the constant. Since we only need the slope, we can get that using [0]

Reply:
Yep that works, we don't need the constant